node/rpcstack_test.go

```diff
@@ -54,25 +54,111 @@ func TestVhosts(t *testing.T) {
 
 // TestWebsocketOrigins makes sure the websocket origins are properly handled on the websocket server.
 func TestWebsocketOrigins(t *testing.T) {
+	tryWebsocketOriginsSimpleRule(t)
+	// tryWebsocketOriginsRuleWithScheme(t)
+	// tryWebsocketOriginsRuleWithPort(t)
+	// tryWebsocketOriginsRuleWithSchemeAndPort(t)
+}
```

Comment:
It would be nice to make this a table-driven test instead, i.e.

```
type originTest struct{
    origin string
    allowSpec string
    shouldAllow bool
}

tests := []originTest{
    {origin: "http://localhost:8080", allowSpec: "localhost", shouldAllow: true},
    ...
}
```

Reply:
That wouldn't be more concise than the status quo, and I think it would be harder to read/follow.

Reply:
I think it would also be slower if spinning up a new server for every test, as the spec is a command-line parameter passed in server startup.